src/mcp_jupyter/server.py

```diff
@@ -395,11 +466,47 @@ def _query_get_position_index(
     # Ensure the notebook path has the .ipynb extension
     notebook_path = _ensure_ipynb_extension(notebook_path)
 
-    # Extract the int version.
+    # Add parameter validation and user guidance
+    if execution_count is not None:
+        # Validate execution_count format and provide helpful guidance
+        try:
+            execution_count_int = _extract_execution_count(execution_count)
+        except ValueError as e:
+            # Get available execution counts for helpful suggestions
+            try:
+                cells_info = _get_available_execution_counts(notebook_path)
+                suggestions = []
+                if cells_info['execution_counts']:
+                    available_counts = ', '.join(map(str, cells_info['execution_counts']))
+                    suggestions.append(f"Available execution_counts: [{available_counts}]")
+                else:
+                    suggestions.append("No cells have been executed yet")
+                
+                suggestions.append(f"Or use position_index (0-{cells_info['total_cells']-1}) instead")
+                suggestion_text = ". " + ". ".join(suggestions) if suggestions else ""
+                
+                raise ValueError(f"{str(e)}{suggestion_text}")
+            except Exception:
+                # If we can't get cell info, just re-raise the original error
+                raise e
```

Comment:
These lines effectively do nothing if we catch the exception and re-raise it so we can safely omit them